src/ui/pages/work/Workorder.js

```diff
@@ -553,6 +531,51 @@ class Workorder extends Entity {
         });
     };
 
+    setClosingCodes = prevState => {
+        const { workorder = {}, layout } = this.state;
+        const { classCode, problemCode, failureCode, causeCode, actionCode, equipmentCode } = workorder || {};
+        const objClass = layout.woEquipment && layout.woEquipment.classCode;
+
+        const { workorder: prevWorkorder = {}, layout: prevLayout } = prevState || {};
+        const prevObjClass = prevLayout && prevLayout.woEquipment && prevLayout.woEquipment.classCode;
+
+        if (layout.reading) {
+            return;
+        }
+
+        if (prevWorkorder.problemCode !== problemCode
+                || prevWorkorder.failureCode !== failureCode
+                || prevWorkorder.causeCode !== causeCode
+                || prevWorkorder.classCode !== classCode
+                || prevWorkorder.equipmentCode !== equipmentCode
+                || prevObjClass !== objClass) {
```

Comment:
what do you think of
const compareByList(argList, a, b) => argList.every(field => a[field] === b[field])
if (!compareByList(["problemCode", "failureCode", "causeCode", "classCode", equipmentCode"], prevWorkorder, workorder)  || prevObjClass !== objClass) {
...
}

Reply:
I think it is clearer. Slightly changed names, please check: https://github.com/cern-eam/eam-light-frontend/pull/114/files#diff-d44e99027fba1cc9226164db6672c2da7be770a7194c16a6bfe83664b8c47171R546